lib/manager/bazel/update.ts

```diff
@@ -114,13 +114,13 @@ export async function updateDependency({
         );
       }
     } else if (
-      (upgrade.depType === 'http_archive' || upgrade.depType === 'http_file') &&
-      upgrade.newValue
+      upgrade.depType === 'http_archive' ||
+      upgrade.depType === 'http_file'
     ) {
       newDef = updateWithNewVersion(
         upgrade.managerData.def,
-        upgrade.currentValue,
-        upgrade.newValue
+        upgrade.currentValue || upgrade.currentDigest,
+        upgrade.newValue || upgrade.newDigest
```

Comment:
Can this mix update version to digest and vice versa?

Reply:
Yes, I think, if e.g., an update object requested a change from a tag to a commit or vice versa.
I think the tag<->commit case would work (Both are archive URLs, though tags seem to be handled as values like releases, while commits, as digests. A similar case seems to be considered for go_repository/git_repository higher up.) But, there'd be problems if the update object tried to request, e.g, a moving a release to a tag (as before)--both in URL format differences and in possibly differing archive extensions.
But regardless, I think it's on the update object to request something that matches, right? When I wrote the code I was thinking it would always be commit->commit, tag->tag, release->release--even if tag<->commit would work. That seems fairly baked into the find-replace code structure.